contracts/gauge-borrowing/VaultPriceOracle.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/math/SafeMathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+
+import "../interfaces/Compound/IPriceOracle.sol";
+import "../interfaces/Compound/IBasePriceOracle.sol";
+import "../interfaces/Compound/ICToken.sol";
+import "../interfaces/Compound/ICErc20.sol";
+import "../interfaces/Compound/IAggregatorV3Interface.sol";
+import "../interfaces/Compound/ILiquidityGauge.sol";
+import "../interfaces/Compound/IRibbonVault.sol";
+import "hardhat/console.sol";
+/**
+ * @title VaultPriceOracle
+ * @notice Returns prices from Chainlink.
+ * @dev Implements `PriceOracle`.
+ * @author David Lucid <david@rari.capital> (https://github.com/davidlucid)
+ */
+contract VaultPriceOracle is IPriceOracle, IBasePriceOracle {
+    using SafeMathUpgradeable for uint256;
+
+    /**
+     * @notice Maps ERC20 token addresses to ETH-based Chainlink price feed contracts.
+     */
+    mapping(address => IAggregatorV3Interface) public priceFeeds;
+
+    /**
+     * @notice Maps ERC20 token addresses to enums indicating the base currency of the feed.
+     */
+    mapping(address => FeedBaseCurrency) public feedBaseCurrencies;
+
+    /**
+     * @notice Enum indicating the base currency of a Chainlink price feed.
+     */
+    enum FeedBaseCurrency {
+        ETH,
+        USD,
+        BTC
+    }
+
+    /**
+     * @notice Chainlink ETH/USD price feed contracts.
+     */
+    IAggregatorV3Interface public constant ETH_USD_PRICE_FEED = IAggregatorV3Interface(0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419);
+
+    /**
+     * @notice Chainlink BTC/ETH price feed contracts.
+     */
+    IAggregatorV3Interface public constant BTC_ETH_PRICE_FEED = IAggregatorV3Interface(0xdeb288F737066589598e9214E782fa5A8eD689e8);
+
+    /**
+     * @dev The administrator of this `MasterPriceOracle`.
+     */
+    address public admin;
+
+    /**
+     * @dev Controls if `admin` can overwrite existing assignments of oracles to underlying tokens.
+     */
+    bool public canAdminOverwrite;
+
+    /**
+     * @dev Constructor to set admin and canAdminOverwrite.
+     */
+    constructor (address _admin, bool _canAdminOverwrite) public {
+        admin = _admin;
+        canAdminOverwrite = _canAdminOverwrite;
+    }
+
+    /**
+     * @dev Changes the admin and emits an event.
+     */
+    function changeAdmin(address newAdmin) external onlyAdmin {
+        address oldAdmin = admin;
+        admin = newAdmin;
+        emit NewAdmin(oldAdmin, newAdmin);
+    }
+
+    /**
+     * @dev Event emitted when `admin` is changed.
+     */
+    event NewAdmin(address oldAdmin, address newAdmin);
+
+    /**
+     * @dev Modifier that checks if `msg.sender == admin`.
+     */
+    modifier onlyAdmin {
```

Comment:
Could we use Ownable from OpenZeppelin instead?  Those contracts are audited.
And using access control roles is even better if you get to that.